metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -89,16 +85,27 @@ final class RenameProvider(
           .topMethodParents(occurence.symbol, semanticDb)
         txtParams <- if (parentSymbols.isEmpty) List(textParams)
         else parentSymbols.map(toTextParams)
-        currentReferences = referenceProvider.references(
-          toReferenceParams(txtParams)
-        )
+        isLocal = occurence.symbol.isLocal
+        currentReferences = referenceProvider
+          .references(
+            // we can't get definition by name for local symbols
+            toReferenceParams(txtParams, includeDeclaration = isLocal),
+            // local symbol will not contain a proper name
+            strictCheck = !isLocal
+          )
+          .locations
+        definitionLocation = if (parentSymbols.isEmpty)
```

Comment:
nit: I always wrap if/else in blocks inside for comprehensions to workaround wonky scalafmt formatting
```suggestion
        definitionLocation = {if (parentSymbols.isEmpty)
        ...
        
        }
```

Reply:
Fixed